### PR TITLE
Add parameter types to docker.models.images.ImageCollection.build

### PR DESCRIPTION
**Pull request by adamtheturtle**
At
https://github.com/docker/docker-py/blob/a3652028b1ead708bd9191efb286f909ba6c2a49/docker/models/images.py#L296 the kwargs are passed through to docker.api.build.BuildApiMixin, so I copied the types from there.

### CODE REVIEW

File: stubs/docker/docker/models/images.pyi [hunk 1]
```diff
@@ -31,7 +42,35 @@ class RegistryData(Model):
 
 class ImageCollection(Collection[Image]):
     model: type[Image]
-    def build(self, **kwargs) -> tuple[Image, Iterator[Any]]: ...
+    def build(
```
Comment (adamtheturtle) [inline stubs/docker/docker/models/images.pyi:45]: Types copied from https://github.com/python/typeshed/blob/f66d687b089f1147f6de1f8dd8142bb514a8d3e6/stubs/docker/docker/api/build.pyi#L21

File: stubs/docker/docker/models/images.pyi [hunk 2]
```diff
@@ -1,10 +1,21 @@
...
+from io import StringIO
+from typing import IO, Any, Literal, TypedDict, overload, type_check_only
 from typing_extensions import TypeAlias
 
 from .resource import Collection, Model
 
 _ImageList: TypeAlias = list[Image]  # To resolve conflicts with a method called "list"
+# Type alias for JSON, explained at:
+# https://github.com/python/typing/issues/182#issuecomment-1320974824.
+_JSON: TypeAlias = dict[str, _JSON] | list[_JSON] | str | int | float | bool | None
```
Comment (adamtheturtle) [inline stubs/docker/docker/models/images.pyi:11]: I'm open to moving this to a common place (as it is duplicated from `docker.utils.json_stream` - but if so, I'd like a suggestion on where that should go / what it should be named, please.

Comment (srittau) [reply stubs/docker/docker/models/images.pyi:11]: We could add a typeshed-only module like `docker._types`.

Comment (adamtheturtle) [reply stubs/docker/docker/models/images.pyi:11]: I will try that and then submit this PR. Thank you!

Comment (adamtheturtle) [reply stubs/docker/docker/models/images.pyi]: @srittau I have done this and marked this PR as ready for review.

File: stubs/docker/docker/_types.pyi [hunk 3]
```diff
@@ -0,0 +1,5 @@
+from typing_extensions import TypeAlias
```
[the remark this answers is not in the document]
Comment (adamtheturtle) [reply stubs/docker/docker/_types.pyi:4]: I have added a comment

**Review by srittau — APPROVED** [review]
Thanks!

**Pull request merged**